lib/vsc/utils/script_tools.py

```diff
@@ -210,3 +226,173 @@ def critical_exception_handler(self, tp, value, traceback):
         message = "Script failure: %s - %s" % (tp, value)
         sys.exc_clear()
         self.critical(message)
+
+
+class CLI(object):
+    """
+    Base class to implement cli tools that require timestamps, nagios checks, etc.
+    """
+    TIMESTAMP_MANDATORY = True
+
+    CLI_OPTIONS = {}
+    CACHE_DIR = "/var/cache"
+
+    def __init__(self, name=None, default_options=None):
+        """
+        Option
+            name (default: script name from commandline)
+            default_options: pass different set of default options
+                (only when creating a new parent class; for regular child classes, use CLI_OPTIONS)
+        """
+        if name is None:
+            name = _script_name(sys.argv[0])
+        self.name = name
+
+        self.fulloptions = self.make_options(defaults=default_options)
+        self.options = self.fulloptions.options
+
+        self.thresholds = None
+
+        self.start_timestamp = None
+        self.current_time = None
+
+    def make_options(self, defaults=None):
+        """
+        Take the default sync options, set the default timestamp file and merge
+        options from class constant OPTIONS
+
+        Return ExtendedSimpleOption instance
+        """
+        if defaults is None:
+            defaults = DEFAULT_CLI_OPTIONS
+        # use a copy
+        options = deepcopy(defaults)
+
+        # insert default timestamp value file based on name
+        if TIMESTAMP_FILE_OPTION in options:
+            tsopt = list(options[TIMESTAMP_FILE_OPTION])
+            tsopt[-1] = os.path.join(self.CACHE_DIR, "%s.timestamp" % self.name)
+            options[TIMESTAMP_FILE_OPTION] = tuple(tsopt)
+
+        options.update(self.CLI_OPTIONS)
+
+        if TIMESTAMP_FILE_OPTION not in options and self.TIMESTAMP_MANDATORY:
+            raise Exception("no mandatory %s option defined" % (TIMESTAMP_FILE_OPTION,))
+
+        return ExtendedSimpleOption(options)
+
+    def _update_kwargs_modified_since(self, kwargs):
```

Comment:
this method is probably only useful in teh AP client

Reply:
Or any other thing that interacts with something?